frontend/src/citizen-frontend/auth/api.ts

```diff
@@ -21,3 +24,44 @@ export type AuthStatus =
 export async function getAuthStatus(): Promise<AuthStatus> {
   return (await client.get<JsonOf<AuthStatus>>('/auth/status')).data
 }
+
+interface WeakLoginRequest {
+  username: string
+  password: string
+}
+
+let nextWeakLoginAttempt = HelsinkiDateTime.now()
+
+export async function authWeakLogin(
+  username: string,
+  password: string,
+  opts: { retryOnRateLimit: boolean } = {
+    retryOnRateLimit: true
+  }
+): Promise<void> {
+  const reqBody: WeakLoginRequest = { username, password }
+
+  // apply client-side rate limit
+  const now = HelsinkiDateTime.now()
+  const sleepMs =
+    nextWeakLoginAttempt.toSystemTzDate().getTime() -
+    now.toSystemTzDate().getTime()
+  if (sleepMs > 0) {
+    await new Promise((resolve) => setTimeout(resolve, sleepMs))
+  }
+
+  try {
+    await client.post('/auth/weak-login', reqBody)
+    return
+  } catch (e) {
+    nextWeakLoginAttempt = HelsinkiDateTime.now().addSeconds(1)
+    if (
```

Comment:
Korvaisin tämän iffin:
```suggestion
    if (isAxiosError(e) && e.response?.status === 429) {
      throw new Error('RATE_LIMITED')
     }
```